src/can.rs

```diff
@@ -0,0 +1,357 @@
+//! CAN bus peripheral control.
+//!
+//! It is called Two-Wire Automotive Interface (TWAI) in ESP32 documentation.
+//!
+
+use crate::delay::{portMAX_DELAY, TickType};
+use crate::gpio::*;
+use core::marker::PhantomData;
+use core::time::Duration;
+use esp_idf_sys::*;
+
+/// CAN timing
+pub enum Timing {
+    B25K,
+    B50K,
+    B100K,
+    B125K,
+    B250K,
+    B500K,
+    B800K,
+    B1M,
+}
+
+impl Timing {
```

Comment:
This is OK, but for consistency with other places, how about you re-work this code so that it is 
`impl From<Timing> for twai_timing_config_t {}`? E.g. see [here]https://github.com/esp-rs/esp-idf-hal/blob/master/src/adc.rs#L92() how this pattern is used across the code.